crates/sui-types/src/messages.rs

```diff
@@ -948,10 +982,70 @@ impl PartialEq for SignedTransaction {
         self.signed_data == other.signed_data && self.auth_sign_info == other.auth_sign_info
     }
 }
+impl Eq for SignedTransaction {}
 
 pub type CertifiedTransaction = TransactionEnvelope<AuthorityStrongQuorumSignInfo>;
 pub type TxCertAndSignedEffects = (CertifiedTransaction, SignedTransactionEffects);
 
+// We do allow serialization/deserialization of this type so that we can write verified data to the
+// db and not re-verify it after loading it. However, be careful not to deserialize this type from
+// an untrusted source such as the network!
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VerifiedTransactionEnvelope<T>(TransactionEnvelope<T>);
```

Comment:
I think that it would be beneficial for us to never impl `Serialize` or `Deserialize` on any "Verified" wrappers because otherwise its possible to read/write these to the wire (which we very much do not want to do). Instead we can ensure that our APIs to the database are carefully constructed such that writes always require a Verified version but internally we serialize the inner type and then reads can use the unchecked versions to return Verified types.

Reply:
good idea, will do